misc/ballerinalang-data-mapper/src/main/java/org/ballerinalang/datamapper/AIDataMapperCodeActionUtil.java

```diff
@@ -104,10 +106,15 @@
         // Insert function call in the code where error is found
         Range newTextRange = CommonUtil.toRange(diagnostic.location().lineRange());
 
-        String filePath = context.filePath().getFileName().toString();
+        Optional<Document> srcFile = context.workspace().document(context.filePath());
+
+        if (srcFile.isEmpty()) {
+            return fEdits;
+        }
+
         LinePosition linePosition = LinePosition.from(context.cursorPosition().getLine(), context.cursorPosition().
                 getCharacter());
-        String symbolAtCursor = semanticModel.symbol(filePath, linePosition).get().name();
+        String symbolAtCursor = semanticModel.symbol(srcFile.get(), linePosition).get().name();
```

Comment:
`srcFile.get()` return a Document. 
But need to get the filePath, 
instead of ``srcFile.get()`, `srcFile.get().syntaxTree().filePath()` this works fine.

Reply:
This is the change done in this PR. Now `symbol()` accepts a `Document` instance instead of a `String` param for the file name.